### PR TITLE
Tech 4449 appraisal phase 1

### PR DESCRIPTION
**Pull request by jebentier**
### Added
- Added support for rails 5 and 6.
- Added appraisal tests for all supported rails version: 4/5/6

### Changed
- Updated various test to be compatible with rails version 4/5/6
- Updated the CI pipeline to test against all three supported versions of rails

### Fixed
- Fixed undefined method `delegate` bug in ActiveSupport version 4

### CODE REVIEW

File: .dependabot/config.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+version: 1
+update_configs:
+- package_manager: "ruby:bundler"
+  directory: "/"
+  update_schedule: "live"
+  version_requirement_updates: "off"
+  commit_message:
+    prefix: "No-Jira"
+    include_scope: true
```
Comment (ColinDKelley) [inline .dependabot/config.yml:10]: Missing newline here I think?

File: .gitignore [hunk 2]
```diff
@@ -51,3 +51,6 @@ build-iPhoneSimulator/
 
 # rubocop
 .rubocop-https---raw-githubusercontent-com-Invoca-style-guide-master-ruby--rubocop-yml
+
+# appraisal
+gemfiles/*.lock
```
Comment (ColinDKelley) [inline .gitignore:56]: Interesting--I'd missed this subtlety. The appraisals generate a fresh `Gemfile.lock` each time?

File: .ruby-version [hunk 3]
```diff
@@ -0,0 +1 @@
+2.6.5
```
Comment (ColinDKelley) [inline .ruby-version:1]: Is the hyphen file best practice?

File: Appraisals [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  gem 'activesupport', '~> 4.2'
+end
+
+appraise 'rails-5' do
+  gem 'activesupport', '~> 5.2'
+end
+
+appraise 'rails-6' do
+  gem 'activesupport', '~> 6.0'
+end
```
Comment (ColinDKelley) [inline Appraisals:13]: :100: So clean!

File: Gemfile [hunk 5]
```diff
@@ -6,18 +6,15 @@ gemspec
...
+gem 'rubocop-git'
+gem 'ruby-prof'
+gem 'ruby-prof-flamegraph'
 
-group :test do
-  gem 'rspec'
-  gem 'rspec_junit_formatter'
-end
+gem 'rspec'
+gem 'rspec_junit_formatter'
```
Comment (ColinDKelley) [inline Gemfile:20]: Thanks for simplifying here!

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,18 @@
 Inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 Note: this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [0.7.0] - Unreleased
```
Comment (ColinDKelley) [inline CHANGELOG.md:6]: BTW I have 0.7.0 also in a branch. We may be able to release those together.

File: spec/lib/contextual_logger/mixins/active_support_tagged_logging_spec.rb [hunk 7]
```diff
@@ -9,9 +9,10 @@
 describe 'ContextualLogger::Overrides::ActiveSupport::TaggedLogging::Formatter' do
   before do
     Time.now_override = Time.now
-    @logger = ContextualLogger.new(Logger.new('/dev/null'))
+    @logger = Logger.new('/dev/null')
     @logger.formatter = ->(_, _, _, msg_with_context) { "#{msg_with_context.to_json}\n" }
     @logger = ActiveSupport::TaggedLogging.new(@logger)
+    @logger = ContextualLogger.new(@logger)
```
Comment (ColinDKelley) [inline spec/lib/contextual_logger/mixins/active_support_tagged_logging_spec.rb:15]: Yuck, I hadn't noticed how this overwrites `@logger`! I will fix that on my branch.

File: .gitignore (hunk 2 above)
Comment (jebentier) [reply .gitignore:56]: yea, this is a best practice set by the appraisals gem to make sure that the appraisal is always tested against the most recent version of the lock.  it leans very heavily on semantic versioning and locking.

File: .ruby-version (hunk 3 above)
Comment (jebentier) [reply .ruby-version:1]: `.ruby_version` isn't picked up by `rbenv` only `.ruby-version` is. so the file name change was necessary for the configuration to take affect.

File: .dependabot/config.yml (hunk 1 above)
Comment (jebentier) [reply .dependabot/config.yml:10]: looks like my script isn't adding the newline to the end 👎 i've gone ahead and added it

**Pull request merged**